File: src/main/java/com/github/gquintana/laapin/joueur/Demo1Joueur.java
package com.github.gquintana.laapin.joueur;

import static com.github.gquintana.laapin.joueur.Action.avancer;
import static com.github.gquintana.laapin.joueur.Action.seReposer;

public class Demo1Joueur implements Joueur {
    @Override
    public Action reflechir(Lapin monLapin, Grille grille) {
        Carotte carotte = grille.carotteProche(monLapin);
        if (carotte == null) {
            return seReposer();
        } else if (carotte.estADroite().de(monLapin)) {
            return avancer().aDroite();

        } else if (carotte.estAGauche().de(monLapin)) {
            return avancer().aGauche();

        } else if (carotte.estEnHaut().de(monLapin)) {
            return avancer().enHaut();

        } else if (carotte.estEnBas().de(monLapin)) {
            return avancer().enBas();
        }
        return null;
    }
}

File: src/main/resources/com/github/mixteen/laapin/joueur/Demo1Joueur.js
var TypeAction = Java.type("com.github.mixteen.laapin.joueur.TypeAction");
var Direction = Java.type("com.github.mixteen.laapin.joueur.Direction");
var carotte = grille.carotteProche(monLapin);
if (carotte) {
    if (carotte.estADroite().de(monLapin)) {
        joueur.avancer().aDroite();
    } else if (carotte.estAGauche().de(monLapin)) {
        joueur.avancer().aGauche();
    } else if (carotte.estEnHaut().de(monLapin)) {
        joueur.avancer().enHaut();
    } else if (carotte.estEnBas().de(monLapin)) {
        joueur.avancer().enBas();
    }
}
